// @fluentui/react-tabster/lib-commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    useArrowNavigationGroup: function() {
        return _index.useArrowNavigationGroup;
    },
    useFocusableGroup: function() {
        return _index.useFocusableGroup;
    },
    useFocusFinders: function() {
        return _index.useFocusFinders;
    },
    useFocusVisible: function() {
        return _index.useFocusVisible;
    },
    useFocusWithin: function() {
        return _index.useFocusWithin;
    },
    useKeyboardNavAttribute: function() {
        return _index.useKeyboardNavAttribute;
    },
    useModalAttributes: function() {
        return _index.useModalAttributes;
    },
    useTabsterAttributes: function() {
        return _index.useTabsterAttributes;
    },
    useObservedElement: function() {
        return _index.useObservedElement;
    },
    useFocusObserved: function() {
        return _index.useFocusObserved;
    },
    useMergedTabsterAttributes_unstable: function() {
        return _index.useMergedTabsterAttributes_unstable;
    },
    useRestoreFocusSource: function() {
        return _index.useRestoreFocusSource;
    },
    useRestoreFocusTarget: function() {
        return _index.useRestoreFocusTarget;
    },
    useUncontrolledFocus: function() {
        return _index.useUncontrolledFocus;
    },
    useOnKeyboardNavigationChange: function() {
        return _index.useOnKeyboardNavigationChange;
    },
    useSetKeyboardNavigation: function() {
        return _index.useSetKeyboardNavigation;
    },
    createCustomFocusIndicatorStyle: function() {
        return _index1.createCustomFocusIndicatorStyle;
    },
    createFocusOutlineStyle: function() {
        return _index1.createFocusOutlineStyle;
    },
    applyFocusVisiblePolyfill: function() {
        return _index1.applyFocusVisiblePolyfill;
    },
    KEYBORG_FOCUSIN: function() {
        return _keyborg.KEYBORG_FOCUSIN;
    },
    TabsterTypes: function() {
        return _tabstertypes601donotuse;
    },
    /** @deprecated Use element.dispatchEvent(new GroupperMoveFocusEvent({ action: GroupperMoveFocusActions.Escape })) */ dispatchGroupperMoveFocusEvent: function() {
        return _tabster.dispatchGroupperMoveFocusEvent;
    },
    /** @deprecated Use element.dispatchEvent(new MoverMoveFocusEvent({ key: MoverKeys.ArrowDown })) */ dispatchMoverMoveFocusEvent: function() {
        return _tabster.dispatchMoverMoveFocusEvent;
    },
    MoverMoveFocusEventName: function() {
        return MoverMoveFocusEventName;
    },
    MoverMoveFocusEvent: function() {
        return MoverMoveFocusEvent;
    },
    MoverKeys: function() {
        return MoverKeys;
    },
    GroupperMoveFocusEventName: function() {
        return GroupperMoveFocusEventName;
    },
    GroupperMoveFocusEvent: function() {
        return GroupperMoveFocusEvent;
    },
    GroupperMoveFocusActions: function() {
        return GroupperMoveFocusActions;
    }
});
const _interop_require_wildcard = require("@swc/helpers/_/_interop_require_wildcard");
const _index = require("./hooks/index");
const _index1 = require("./focus/index");
const _tabster = require("tabster");
const _keyborg = require("keyborg");
const _tabstertypes601donotuse = /*#__PURE__*/ _interop_require_wildcard._(require("./tabster-types-6.0.1-do-not-use"));
const MoverMoveFocusEventName = _tabster.Events.MoverMoveFocusEventName;
const MoverMoveFocusEvent = _tabster.Events.MoverMoveFocusEvent;
const MoverKeys = _tabster.Types.MoverKeys;
const GroupperMoveFocusEventName = _tabster.Events.GroupperMoveFocusEventName;
const GroupperMoveFocusEvent = _tabster.Events.GroupperMoveFocusEvent;
const GroupperMoveFocusActions = _tabster.Types.GroupperMoveFocusActions;
